vital.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+#include <thread>
+#include <chrono>
+#include <string>
+#include <set>
+#include "./vital.h"
+
+using std::cout;
+using std::wcout;
+using std::flush;
+using std::this_thread::sleep_for;
+using std::chrono::seconds;
+
+Vital::Vital(float vitalValue, const VitalBaseline& baseline) :
+    value(vitalValue), vitalbase(baseline) {
+}
+
+std::set<int>& Vital::collectInvalidCategories() const {
+    static auto numCategories{ static_cast<int>(vitalbase.levels.size()) };
+    static std::set<int> invalidCategoies{ 0, 1, numCategories - 1, numCategories };
+    return invalidCategoies;
+}
+
+int Vital::getCategory() const {
+    auto iter = std::lower_bound(vitalbase.levels.begin(), vitalbase.levels.end(), value);
+    if (value == vitalbase.low) {
+        return static_cast<int>(std::distance(vitalbase.levels.begin(), iter)) + 1;
+    } else {
+        return static_cast<int>(std::distance(vitalbase.levels.begin(), iter));
+    }
+}
+
+void Vital::displayMessage(std::wstring message) const {
+    wcout << message << std::endl;
+}
+
+void Vital::displayTransitionGraphics() const {
```

Comment:
This function is your mother-in-law :)
It's there but you don't like it, so you will not call it

Reply:
:)..
sorry, I commented it for testing. Will add it back